Add JSDoc documentation

timmathews opened the pull request:
A first pass at documentation

timmathews committed:
Add JSDoc documentation

tkurki commented:
I think this was there beforehand, but should be fixed:
```
tjk:signalk-js-client tjk$ npm run doc

> signalk-client@0.0.1-4 doc /Users/tjk/git-workspace/signalk/signalk-js-client
> jsdoc -r lib

ERROR: Unable to find the source file or directory /Users/tjk/git-workspace/signalk/signalk-js-client/lib
There are no input files to process.
```
How do you publish jsdoc in Github?

timmathews commented:
I think we would need to have  a gh-pages branch configured to publish to signalk.org/signalk-js-client and add a doc:publish script to automate the process. Not sure about styling to make it look like gitbook.

tkurki commented:
I would rather keep the docs right next to the readme and code in Github proper, not a separate web page.

How about https://github.com/jsdoc2md/jsdoc-to-markdown ?

timmathews commented:
That'll work

timmathews committed:
Convert design.md to JSDoc on stubbed index.js

timmathews committed:
Write documentation to README.md

timmathews committed:
Latest versions of dependencies

timmathews committed:
Implement Constructor

Add constructor functionality with tests

timmathews committed:
Stub tests for connect function

timmathews committed:
Clean up package.json

timmathews committed:
Implement Client#connect

Client#connect and associated tests.

TODO: Create skConnection object with all properties

tkurki commented on index.js:
```diff
@@ -129,7 +150,48 @@ function Client(options) {
+      });
+    }
+
+    const ws = new WebSocket(this.wsPath);
```
This will now always open the ws connection?

tkurki commented on index.js:
```diff
@@ -129,7 +150,48 @@ function Client(options) {
+    ws.onclose = onClose;
+    ws.onerror = onError;
+
+    ws.send(JSON.stringify({context: 'vessels.self', subscribe: paths}));
```
The connection is not yet open i think.

tkurki commented:
What's the plan for this PR? It is not just jsdoc, but also a bunch of tests and conversion to ES6. And the diff for index.js looks....challenging for a reviewer.

timmathews commented:
This branch has diverged *a lot* from the initial intention. Closing the PR and renaming the branch.
